tests/e2e/network/ensureloadbalancer.go

```diff
@@ -649,62 +649,84 @@ var _ = Describe("Ensure LoadBalancer", Label(utils.TestSuiteLabelLB), func() {
 	})
 
 	It("should support node label `node.kubernetes.io/exclude-from-external-load-balancers`", func() {
-		if os.Getenv(utils.AKSTestCCM) == "" {
-			Skip("Skip this test case for non-AKS test")
-		}
-
 		label := "node.kubernetes.io/exclude-from-external-load-balancers"
 		By("Checking the number of the node pools")
 		nodes, err := utils.GetAgentNodes(cs)
 		Expect(err).NotTo(HaveOccurred())
-		initNodepoolNodeMap := utils.GetNodepoolNodeMap(&nodes)
-		if len(initNodepoolNodeMap) != 1 {
-			Skip("single node pool is needed in this scenario")
+		if os.Getenv(utils.AKSTestCCM) != "" {
+			// AKS
+			initNodepoolNodeMap := utils.GetNodepoolNodeMap(&nodes)
+			if len(initNodepoolNodeMap) != 1 {
+				Skip("single node pool is needed in this scenario")
+			}
+		}
+		lbBPConfigTypeNodeIP := false
+		if os.Getenv(utils.LBBackendPoolConfigType) == "nodeIP" {
+			lbBPConfigTypeNodeIP = true
 		}
 
 		By("Creating a service to trigger the LB reconcile")
 		service := utils.CreateLoadBalancerServiceManifest(testServiceName, nil, labels, ns.Name, ports)
 		_, err = cs.CoreV1().Services(ns.Name).Create(context.TODO(), service, metav1.CreateOptions{})
 		Expect(err).NotTo(HaveOccurred())
 		publicIPs, err := utils.WaitServiceExposureAndValidateConnectivity(cs, tc.IPFamily, ns.Name, testServiceName, []string{})
+		defer func() {
+			deleteSvcErr := utils.DeleteService(cs, ns.Name, testServiceName)
+			Expect(deleteSvcErr).NotTo(HaveOccurred())
+		}()
 		Expect(err).NotTo(HaveOccurred())
 		Expect(len(publicIPs)).NotTo(BeZero())
 		publicIP := publicIPs[0]
 
 		By("Checking the initial node number in the LB backend pool")
 		lb := getAzureLoadBalancerFromPIP(tc, publicIP, tc.GetResourceGroup(), "")
-		lbBackendPoolIPConfigsCount := 0
-		if utils.IsAutoscalingAKSCluster() {
-			for _, ipconfig := range *(*lb.BackendAddressPools)[getLBBackendPoolIndex(lb)].BackendIPConfigurations {
-				if !strings.Contains(*ipconfig.ID, utils.SystemPool) {
-					lbBackendPoolIPConfigsCount++
+		if lbBPConfigTypeNodeIP {
+			lbBackendPoolIPCount := len(*((*lb.BackendAddressPools)[getLBBackendPoolIndex(lb)].LoadBalancerBackendAddresses))
+			Expect(lbBackendPoolIPCount).To(Equal(len(nodes)))
+			utils.Logf("Initial node number in the LB backend pool is %d", lbBackendPoolIPCount)
+		} else {
+			lbBackendPoolIPConfigsCount := 0
+			if utils.IsAutoscalingAKSCluster() {
+				for _, ipconfig := range *(*lb.BackendAddressPools)[getLBBackendPoolIndex(lb)].BackendIPConfigurations {
```

Comment:
We can check loadBalancerBackendAddresses as well for nic lb

Reply:
I think it is a good idea but after I tried, it seems not possible.
I found that the LoadBalancerBackendAddress struct in:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/0e6c0a55c732dad2e4b8b946b234c118259769f4/vendor/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2022-07-01/network/models.go
is:
```
type LoadBalancerBackendAddress struct {
	// LoadBalancerBackendAddressPropertiesFormat - Properties of load balancer backend address pool.
	*LoadBalancerBackendAddressPropertiesFormat `json:"properties,omitempty"`
	// Name - Name of the backend address.
	Name *string `json:"name,omitempty"`
}
```
It should have ID but actually not. ID is need for autoscaling cluster to avoid taking systempool into consideration.

Reply:
it's in properties.networkInterfaceIPConfiguration

Reply:
Thank you. Updated